share/cmake/modules/FindExtPackages.cmake

```diff
@@ -105,20 +89,37 @@ if(OCIO_BUILD_PYTHON OR OCIO_BUILD_DOCS)
     endif()
 endif()
 
-# The presence of OpenImageIO allows additional OCIO apps and the OSL
-# translation unit tests to be built.
+# The presence of OpenImageIO or OpenEXR allows additional OCIO apps to be built.
 
 if(OCIO_BUILD_APPS OR OCIO_BUILD_TESTS)
+    # Try to use OpenImageIO first as it supports more formats.
+
     # OpenImageIO
     # https://github.com/OpenImageIO/oiio
-
     set(OIIO_VERSION "2.1.9")
 
     if(OCIO_USE_OIIO_CMAKE_CONFIG)
         find_package(OpenImageIO ${OIIO_VERSION} CONFIG)
     else()
         find_package(OpenImageIO ${OIIO_VERSION})
     endif()
+
+    if(OpenImageIO_FOUND AND TARGET OpenImageIO::OpenImageIO)
+        add_library(OpenColorIO::ImageIOBackend ALIAS OpenImageIO::OpenImageIO)
+        set(OCIO_IMAGE_BACKEND OpenImageIO)
+    else()
+        # OpenEXR
+        # https://github.com/AcademySoftwareFoundation/openexr
+        find_package(OpenEXR 3.1.5)
+
+        if(TARGET OpenEXR::OpenEXR)
+            add_library(OpenColorIO::ImageIOBackend ALIAS OpenEXR::OpenEXR)
+            set(OCIO_IMAGE_BACKEND OpenEXR)
+        endif()
+    endif()
+
+    message(STATUS "Using ${OCIO_IMAGE_BACKEND} to build ociolutimage, ocioconvert and ociodisplay.")
```

Comment:
So the logic here is that if OIIO is found, it will use it.

I wonder if a better behavior would be to default to using OpenEXR (even in the presence of OIIO), but having a build option that allows people to select OIIO.

Remember, a longstanding goal is to remove the circular dependency. I worry that for the common case of people having both packages on their system, this logic will preserve the circular dependency and that might come back to bite them.

Imagine the following scenario: An older OIIO and OCIO are already on the system. They desire to build a new OCIO, followed by a new OIIO and uses the new OCIO. But here what will happen is that they'll end up with an OCIO that builds against the old OIIO they no longer want to use. Is that problematic? There's no easy way, short of un-installing OIIO ahead of building OCIO, to ensure that the OCIO build doesn't use the OIIO that they don't want.

My recommendation would be to make the circular dependency entirely "opt-in." Or at the very least, make an option that allows them to easily opt out of the circular dependency, even if OIIO can be found.

Reply:
> My recommendation would be to make the circular dependency entirely "opt-in." Or at the very least, make an option that allows them to easily opt out of the circular dependency, even if OIIO can be found

Yes please!

Reply:
Thanks for the feedback, I agree with all being said. There is now a CMake option called `OCIO_USE_OIIO_FOR_APPS` to explicitly opt-in to use OIIO, otherwise it will default on OpenEXR.